src/aosm/azext_aosm/generate_nsd/nsd_generator.py

```diff
@@ -131,14 +162,23 @@ def write_nf_bicep(self) -> None:
         bicep_deploymentValues = ""
 
         deploy_parameters_dict = json.loads(self.deploy_parameters)
-        deploy_properties = deploy_parameters_dict["properties"]
-
-        for key, value in deploy_properties.items():
-            # location is sometimes part of deploy_properties.
-            # We want to avoid having duplicate params in the bicep template
-            if key != "location":
-                bicep_params += f"param {key} {value['type']}\n"
-            bicep_deploymentValues += f"{key}: {key}\n  "
+        if "properties" in deploy_parameters_dict:
```

Comment:
I prefer to do this check the other way around, just to avoid indenting too much.
```
if "properties" not in deploy_parameters_dict:
    raise error

do_mainline_stuff .........
```